### PR TITLE
Generalize JOSS to better handle multiple instances

### PR DESCRIPTION
**Pull request by parrish**
##### This looks like a larger PR than it really is since there's a fair bit of template reworking.  It's probably fairly digestable per-commit, though I'm happy to split this out into multiple PR's if it's easier.
----

Motivated by #329, this PR

- Loads the editor team from GitHub for the reviewer assignment and suggested reviewer on the submission.
- Generalizes the templates to use a [settings.yml](https://github.com/openjournals/joss/blob/ab305ef8c266dda60b33f3931c025d8ba9588a26/config/settings.yml) file to specify the easily extractable language and configuration
- Extracts as much customizable content from templates and template structure as possible.  These partials now live in [app/views/content](https://github.com/openjournals/joss/blob/ab305ef8c266dda60b33f3931c025d8ba9588a26/app/views/content)
- Adds a simple editors admin management UI at [http://staging.joss.theoj.org/editors](http://staging.joss.theoj.org/editors)
  - Can be bootstrapped via the [db/seeds](https://github.com/openjournals/joss/blob/ab305ef8c266dda60b33f3931c025d8ba9588a26/db/seeds.rb)
  - Currently, it's independent from the GitHub editor team, but that might be a cool feature.
  - It requires the admin flag on a user account, just ping me if you can't access it
- Adds a configurable list of [paper submission types](https://github.com/openjournals/joss/blob/ab305ef8c266dda60b33f3931c025d8ba9588a26/config/settings.yml#L20).
  - Each paper type will render a correspondingly named review checklist.
  - e.g. papers of type "awesome" would expect a checklist file at [app/views/content/github/_awesome_review_checklist.erb](https://github.com/openjournals/joss/blob/ab305ef8c266dda60b33f3931c025d8ba9588a26/app/views/content/github/_awesome_review_checklist.erb)

Along with openjournals/whedon-api#11, this should get us into a pretty good position to have multiple journals running. 🎉 

In-progress deploys are going to http://staging.joss.theoj.org

Feedback welcome!

### CODE REVIEW

**Comment by vsoch** [comment]
Huge +1 to this work! I started customizing the joss repo and the majority of "manual changes" were changing the content of titles, etc. Has there been discussion about some form of documentation that details development --> deployment? I didn't have familiarity with the development stack but narrowed it down to (after bundling and dealing with dependencies) these steps:

---------------------------------------------------------------------------------------------------------
## Development
This is an application friendly to run on heroku. So you should first [make an account](https://devcenter.heroku.com/) and then [follow the instructions to deploy the application](https://devcenter.heroku.com/articles/getting-started-with-ruby#introduction). It comes down to:

1. Login with `heroku login`
2. Create the heroku branch with `heroku create`

```
heroku create
Creating app... done, ⬢ shielded-mountain-65855
https://shielded-mountain-65855.herokuapp.com/ | https://git.heroku.com/shielded-mountain-65855.git
```

3. Migrate the database to create the tables, etc.

```
heroku run rake db:migrate
```

Has anyone talked about a Dockerized development server setup?  I have something almost working and can do a PR when this gets merged, with some instructions for getting it up and running!

File: app/views/shared/review_body.erb [hunk 1]
```diff
@@ -7,54 +7,25 @@
...
-- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
-- [ ] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
-- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support
-
-### Software paper
-
-- [ ] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
-- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
-- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
+<%- if paper.kind.present? %>
```
Comment (arfon) [inline app/views/shared/review_body.erb]: Not sure I completely understand this syntax...

Comment (arfon) [reply app/views/shared/review_body.erb]: Ah, it seems like it's the same as `exists?`

**Comment by arfon** [comment]
Closing in favour or #361

**Pull request closed without merging**